datastream-server/src/main/java/com/linkedin/datastream/server/zk/ZkAdapter.java

```diff
@@ -867,29 +908,56 @@ public void cleanupDeadInstanceAssignments(List<String> instances) {
         _liveTaskMap.remove(instance);
       }
     }
+    LOG.debug("unusedTasks remaining after dead instances cleanup: {}", unusedTasks.stream()
+        .map(DatastreamTask::getDatastreamTaskName)
+        .collect(Collectors.toList()));
+    // Clean the remaining unusedTasks.
```

Comment:
Can you add a log before this so that we can find out how long the previous phase of cleanup took? I am concerned that we are increasing the time it will take to cleanup the dead instances, which can lead to handleDoLeaderAssignmentChange failures that can wedge the cluster into a bad state requiring a cluster restart. We should have sufficient logging to understand how long each phase takes in case we need to make a call later on about preventing such stages from getting interrupted in the first place.

Reply:
We are not really increasing the dead instance cleanup time. Just doing the two cleaning process interleaved. I don't really see a value of another debug line in the middle. If we come across a condition when we need it, we can always enable debug logs.

Reply:
I know we have run into issues when cleaning up dead instances gets interrupted and I would prefer having a log to know how long that takes. Cleaning up two nodes instead of one can essentially double the time this process takes since you need to make two calls to ZK to delete.